django/db/migrations/recorder.py

```diff
@@ -71,7 +81,10 @@ def ensure_schema(self):
     def applied_migrations(self):
         """Return a set of (app, name) of applied migrations."""
         if self.has_table():
-            return {tuple(x) for x in self.migration_qs.values_list('app', 'name')}
+            return {
+                AppliedMigration([m.app, m.name], applied=m.applied)
+                for m in self.migration_qs.iterator()
```

Comment:
It don't think it's worth using `iterator()` here given the small size of the result set and the fact it will be fully materialized. Also you could keep using `values_list('app', 'name', 'applied')`.

```python
class AppliedMigration(tuple):
    def __init__(self, app, name, applied):
        instance = tuple.__new__(cls, (app, name))
        instance.applied = applied
        return instance

return {AppliedMigration(x) for x in self.migration_qs.values_list('app', 'name', 'applied')
```

Reply:
Sounds good.